msteams-platform/toolkit/toolkit-v4/debug-your-Teams-app-test-tool-vs.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Debug bot using Teams App Test Tool Visual Studio
+author: v-ganr
+description: Learn how to emulate the Teams experience for your bot in Teams App Test Tool in Visual Studio.
+ms.localizationpriority: medium
+ms.topic: Teams App Test Tool overview
+ms.author: surbhigupta
+ms.date: 02/13/2024
+---
+
+# Teams App Test Tool for Visual Studio
+
+Teams App Test Tool (Test Tool) makes debugging bot-based apps effortless. You can chat with your bot and see its messages and Adaptive Cards as they appear in Microsoft Teams. You don’t need a Microsoft 365 developer account, tunneling, or Teams app and bot registration to use Test Tool.
+
+The following image shows a sample app displaying a response in Test Tool:
+
+ :::image type="content" source="../../assets/images/teams-toolkit-v2/teams-toolkit-vs/test-tool-sample-app-output-vs.png" alt-text="Screenshot shows the Teams App Test Tool Adaptive Card." lightbox="../../assets/images/teams-toolkit-v2/teams-toolkit-vs/test-tool-sample-app-output-vs.png"::: 
+
+The following are the advantages of Test Tool:
+
+* **Sandbox environment**: The sandbox environment of Test Tool emulates the behavior, look, and user experience of Teams.
+
+* **Tunneling**: An external tunnel service isn't necessary as Test Tool runs on a local server that your bot can communicate with.
+
+* **Reduce account dependencies**: Microsoft 365 Developer tenant and the app uploading permissions aren't necessary to debug the bot.
+
+* **Rapid inner-loop iterations**: Optimizes the process of making changes to the app design and bot logic without having to redeploy the bot to the cloud.
+
+* **Mock data and activities**: Test Tool makes it easy to test complex scenarios such as, sending a welcome message when a new member joins the channel, using mock data and activity triggers.
+
+* **Reliable**: Test Tool is reliable as the bot's Adaptive Card utilizes the same rendering technology as in Teams.
+
+* **Integration with existing Teams bot applications**: Test Tool integrates effortlessly with existing Teams bot applications built with Bot Framework SDK.
+
+* **Support for different scopes**: Test Tool supports testing in personal, team, and group chat scopes.
+
+## Prerequisites
+
+Ensure you install the following tools for building and deploying your bots in Test Tool:
+
+| Install | For using... |
+| --- | --- |
+| [Node.js](https://nodejs.org/en/download/) | Back-end JavaScript runtime environment. |
+| [Visual Studio 2022](https://visualstudio.microsoft.com)<br> | You can install the enterprise version in Visual Studio 2022, and install the ASP.NET and web development workloads. Use the version 17.9 Preview 3.0 or later. |
+| [OpenAI](https://platform.openai.com/apps) or  [Azure OpenAI](https://oai.azure.com/portal)| First create your OpenAI API Secret key to use OpenAI's GPT. If you want to host your app or access resources in Azure, you must create an Azure OpenAI service.|
```

Comment:
OpenAI or Azure OpenAI is not a prerequsite to debug bot in Test Tool. Instead, they are the prerequsite to debug AI chat bot.

Reply:
Updated.